addon/components/frost-action-bar.js

```diff
@@ -66,10 +75,44 @@ export default Component.extend({
   @computed('selectedItems.[]')
   selectedItemsLabel (selectedItems) {
     return this.i18n.formatSelectedItemsLabel(selectedItems.length)
-  }
+  },
+
+  @readOnly
+  @computed('controls')
+  isControlsObject (controls) {
+    return typeOf(controls) === 'object'
+  },
+
+  @readOnly
+  @computed('isVisible', 'isControlsObject', 'controls', 'selectedItems.[]')
+  _controls (isVisible, isControlsObject, controls, selectedItems) {
+    if (isVisible && isControlsObject && isPresent(controls) && isPresent(selectedItems)) {
```

Comment:
Why are we checking `isVisible`, `controls` and `selectedItems` for the object but not for the `controls`?

Reply:
I'm not sure that I understand what you mean. I realized after looking at this that I don't need the `isPresent(controls)` check because the `isControlsObject` computed property will not return `'object'` if it's undefined, so I can remove that. `isVisible` is checked because I don't want this computed property to run if the action bar is not visible. The main point of this computed property is to either return the list of applicable controls or just return the same controls list that is passed to the component - for backward compatibility.

Reply:
I'm simply not sure to understand why we are doing all these checks only in the new case. I'm expecting that the same checks are valid for the new use case and the backward compatibility.

Reply:
I see what you are saying now. The reason why I only added them for the new case is because they were not there before, so I assume that we didn't need them for the old case. The `isVisible` is not necessary, so I can remove that. It's not even a computed property that I added, and I did a search project-wide and didn't see any other code that's using it, so it's possible that it's obsolete now. The `isControlsObject` property is the only one that's absolutely necessary. The only reason I added the `isPresent(selectedItems)` for the new case is so that we wouldn't be doing any unnecessary processing on an empty list, but that check can be removed also.